Firestore/core/test/firebase/firestore/core/query_test.cc

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2018 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "Firestore/core/src/firebase/firestore/core/query.h"
+
+#include "Firestore/core/src/firebase/firestore/model/document.h"
+#include "Firestore/core/src/firebase/firestore/model/resource_path.h"
+#include "Firestore/core/test/firebase/firestore/testutil/testutil.h"
+#include "gtest/gtest.h"
+
+namespace firebase {
+namespace firestore {
+namespace core {
+
+using model::Document;
+using testutil::Doc;
+
+TEST(QueryTest, MatchesBasedOnDocumentKey) {
+  Document doc1 = Doc("rooms/eros/messages/1", /*version=*/0);
```

Comment:
None of the other clients use this decoration for versions (or even for all integer parameters).

As far as I know, the advice to name the parameter like this applies to booleans only because they have no significance otherwise.

Reply:
So the production code already solved this problem by using SnapshotVersion::None for this case. The tests are looking for increased clarity by reducing the amount of noise required to get things going.

Since a document is in essence a tuple of path, version, and contents, and they're ubiquitous in our tests, I'd argue that the minor benefit in clarity about the argument is outweighed by the increased verbosity throughout.

Another path could be to make Doc default the argument to zero. We couldn't do this in C but it's an option now.

Reply:
Ok, I'm convinced. But I've also taken your suggestion (default to 0) so it's also now obsolete. :)